### PR TITLE
fix(curriculum): check if count gets incremented by function

### LINKED ISSUE #41535: Redux: Register a Store Listener challenge passes with incorrect answer
**Describe your problem and how to reproduce it:**
Hi there! 

A forum user in this [discussion](https://forum.freecodecamp.org/t/got-a-build-error-but-it-pased-the-test-redux-register-a-store-listener/451922) wanted to know why they received an error message for their code when it passed the test. 

Here are the FCC instructions:
Write a callback function that increments the global variable count every time the store receives an action, and pass this function in to the store.subscribe() method. 

But the user's answer did not include a callback function as you can see in the screenshot.

I think the test should only pass if a callback function was used. 

Thanks! 

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/front-end-libraries/redux/register-a-store-listener

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**
Screen Shot 2021-03-19 at 7 51 59 AM

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #41535

### CODE REVIEW

**Comment by ShaunSHamilton** [comment]
Hey @Sembauke . Probably not a problem now, but this branch is way behind `upstream/main`.

**Comment by Sembauke** [comment]
> Hey @Sembauke . Probably not a problem now, but this branch is way behind `upstream/main`.

Hey Shaun I'll make sure to update my main branch next time 😄

**Review by moT01 — COMMENTED** [review]
I don't know how thorough we want to be here. But I tried `store.subscribe(() => count += 1);` with the implicit return and it doesn't pass since it is checking for the `{`. I came up with this if you want to use it: `store\s*\.\s*subscribe\s*\(\s*(function\s*\(\s*\)|\(\s*\)[ \t]*=>)` Probably still misses something, but it does get the implicit return and allows for spaces in a few other spots.

I took a quick look at [our guide article](https://forum.freecodecamp.org/t/freecodecamp-challenge-guide-register-a-store-listener/301446). It shows how to put the function into a variable and just pass that to `subscribe`. Not sure how we would account for that. Looking around on the forum, there is a number of people using that method, so we could maybe try to account for it. Not sure how though - if it comes to it, I think I would rather allow those to pass than fail because we wanted to fix this issue.

**Comment by Sembauke** [comment]
we also could just look for `()  =>`  and `function`  this seems the easiest way to me to check if the user uses a function.

also adding checks for white space between `store\s*.\s*`  makes the test "There should be a listener function subscribed to the store using store.subscribe."  fail because the test looks directly if `'store.subscribe(')` exists.

**Comment by moT01** [comment]
Yea, I saw enough forum posts of people using that method that I think we need to account for it. Just checking for those globally would be fine with me - I can't think of any other way.

**Review by moT01 — DISMISSED** [review]
There wasn't really any discussion on the issue about this. My only concern is that I'm wondering if it's worth fixing. The particular issue is likely exceptionally rare. With this PR, it is still possible, however even more rare to the point that it will likely never happen, to produce the same issue...
Screen Shot 2021-04-22 at 9 54 37 AM

All that said, I don't think it's hurting anything to add this, so I would be fine adding it. A second opinion would be nice.

LGTM 🎉

**Comment by Sembauke** [comment]
I think with this extra test we made it more clear what the user needs to do. The case you are describing seems like it would never happen.

**Comment by raisedadead** [comment]
Hi @Sembauke we are reserving request for review from dev-team for mostly automated tags as per the code ownership. This being a regular PR for curriculum can be merged without approval requirement from dev-team.

File: curriculum/challenges/english/03-front-end-libraries/redux/register-a-store-listener.md
Comment (gikf) [inline curriculum/challenges/english/03-front-end-libraries/redux/register-a-store-listener.md:38]:
```suggestion
The listener function `store.subscribe` should have a function. 
```
Nitpicking with the dot.

 I'm also having troubles with wrapping my mind around what actually this sentence is saying. If calling `store.subscribe` a listener function is correct, it might be better to still avoid using it. To not confuse it with passed in listener function. It doesn't look great, but maybe something like below?
```suggestion
The `store.subscribe` should receive a function.
```

**Review by gikf — COMMENTED** [review]
PR surely cuts down number of solutions that shouldn't pass, including example from which originated the issue.

Some wording doubt in specific comment.

**Review by gikf — APPROVED** [review]
LGTM :+1:

**Review by moT01 — APPROVED** [review]
LGTM @Sembauke 🎉 Thanks for contributing to freeCodeCamp 🎉

**Comment by Sembauke** [comment]
> LGTM @Sembauke 🎉 Thanks for contributing to freeCodeCamp 🎉

Thanks for reviewing my PR Tom

**Pull request merged**
